packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { useEffect } from 'react';
+import styled, { keyframes } from 'styled-components';
+import { Input, Button, Textarea, Select, variables } from '@trezor/components';
+import { Card, Translation } from '@suite-components';
+import { useForm, Controller } from 'react-hook-form';
+import { ChildProps as Props } from '../../Container';
+import { useActions } from '@suite-hooks';
+import * as signVerifyActions from '@wallet-actions/signVerifyActions';
+import * as modalActions from '@suite-actions/modalActions';
+
+const AppearAnimation = keyframes`
+ 0% { opacity: 0; }
+ 10% { opacity: 0.1; }
+ 30% { opacity: 0.3; }
+ 70% {  opacity: 0.7;}
+ 80% {  opacity: 0.8;}
+ 90% {  opacity: 0.9;}
+ 100% { opacity: 1; }
+`;
+
+const Column = styled.div`
+    display: flex;
+    flex: 1 1 auto;
+    flex-direction: column;
+    animation-name: ${AppearAnimation};
+    animation-duration: 0.6s;
+    animation-iteration-count: 1;
+    @media screen and (max-width: ${variables.SCREEN_SIZE.MD}) {
+        flex: 1 1 100%;
+    }
+`;
+
+const StyledSelect = styled(Select)`
+    padding: 0;
+    margin-bottom: 0px;
+    @media screen and (max-width: ${variables.SCREEN_SIZE.SM}) {
+        font-size: ${variables.FONT_SIZE.TINY};
+    }
+`;
+
+const Row = styled.div`
+    padding-bottom: 28px;
+    display: flex;
+`;
+const RowButtons = styled(Row)`
+    display: flex;
+    align-items: center;
+    justify-content: flex-end;
+`;
+const StyledButton = styled(Button)`
+    width: 110px;
+    margin-left: 10px;
+
+    &:first-child {
+        margin-left: 0;
+    }
+`;
+const LeftSideButton = styled(StyledButton)`
+    margin-right: auto;
+`;
+
+interface AddressToSign {
+    address: string;
+    path: string;
+    accountsSelectOpt?: Array<{ label: string; value: string }>;
+}
+type errorsMessage = 'TR_ADDRESS_IS_NOT_VALID' | 'TR_REQUIRED_FIELD';
+type Inputs = {
+    address: string;
+    path: string;
+    signMessage: string;
+    accountsSelect: { label: string; value: string; type: '' };
+    signature: string;
+};
+
+let path: AddressToSign['address'] = ''; // fast click on copy button failed if put this inside a component
+let address: AddressToSign['path'] = '';
+let accountsSelectOpt: AddressToSign['accountsSelectOpt'];
```

Comment:
`path`, `address` and `accountsSelectOpt` should be a part of `SignMessage` component state (useState hook)
keeping them as out-of-scope variables is anti pattern

In fact the only thing you need to work with is a `Select` currently selected option (no need to split it into 3 separated variables)

Reply:
OK, anti-pattern, I've been writing in react since may 25, less than month, i will fix it